gtda/homology/simplicial.py

```diff
@@ -612,3 +613,202 @@ def transform(self, X, y=None):
         Xt = _postprocess_diagrams(Xt, self._homology_dimensions,
                                    self.infinity_values_, self.n_jobs)
         return Xt
+
+
+@adapt_fit_transform_docs
+class FlagPersistence(BaseEstimator, TransformerMixin):
+    """`Persistence diagrams <https://giotto.ai/theory>`_ resulting from
+    `filtrations of directed or undirecte flag complexes
+    <https://giotto.ai/theory>`_.
+
+    Given a `point cloud <https://giotto.ai/theory>`_ in Euclidean space,
+    or an abstract `metric space <https://giotto.ai/theory>`_ encoded by a
+    distance matrix, information about the appearance and disappearance of
+    topological features (technically, `homology classes
+    <https://giotto.ai/theory>`_) of various dimensions and at different
+    scales is summarised in the corresponding persistence diagram.
+
+    Parameters
+    ----------
+    homology_dimensions : iterable, optional, default: ``(0, 1)``
+        Dimensions (non-negative integers) of the topological features to be
+        detected.
+
+    directed : bool, optional, default: ``True``
+        If true, computes the directed flag complex. Otherwise it
+        computes the undirected flag complex.
+
+    coeff : int prime, optional, default: ``2``
+        Compute homology with coefficients in the prime field
+        :math:`\\mathbb{F}_p = \\{ 0, \\ldots, p - 1 \\}` where
+        :math:`p` equals `coeff`.
+
+    max_entries : int, optional, default: ``-1``
+        Number of entries above which all cells creating columns in the
+        reduction matrix are skipped. Use this for hard problems, a good
+        value is often `100,000`. Increase for higher precision, decrease
+        for faster computation. A negative value computes highest possible
+        precision.
+
+    max_edge_length : float, optional, default: ``numpy.inf``
+        Upper bound on the maximum value of the Vietoris-Rips filtration
+        parameter. Points whose distance is greater than this value will
+        never be connected by an edge, and topological features at scales
+        larger than this value will not be detected.
+
+    infinity_values : float or None, default : ``None``
+        Which death value to assign to features which are still alive at
+        filtration value `max_edge_length`. ``None`` has the same behaviour
+        as `max_edge_length`.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    See also
+    --------
+    ConsistentRescaling
+
+    Notes
+    -----
+    `Flagser <https://github.com/luetge/flagser>`_ is used as a C++ backend
+    for computing Flag complexes persistent homology. Python bindings rely on
+    the `pyflagser <https://github.com/giotto-ai/pyflagser>`_ package.
```

Comment:
What about putting forward the binding, *which  uses `Flagser` package*? No strong feelings for this from my side though.
For example, something along the lines of:
This is an adapter for `pyflagser <https://github.com/giotto-ai/pyflagser>`_, the Python bindings for `Flagser <https://github.com/luetge/flagser>`_ , the C++ backend for computing Flag complexes persistent homology.

Reply:
I have updated the description.